File: 560-SubarraySumEqualsK/560-SubarraySumEqualsK.cpp
// Last updated: 6/11/2026, 11:32:30 AM
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int n = nums.size();
        unordered_map<int, int> ump;
        int ps = 0;
        int ans = 0;
        for(int i=0; i<n; i++){
            ps += nums[i];
            if(ps == k){
                ans++;
            }
            if(ump.find(ps-k)!=ump.end()){
                ans += ump[ps-k];
            }
            ump[ps]++;
        }

        return ans;
    }
};

File: 560-SubarraySumEqualsK/560-SubarraySumEqualsK.py
# Last updated: 6/11/2026, 11:32:33 AM
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        prefixmap = {}
        sum1 = ct = 0
        for i in nums:
            sum1 += i
            if(sum1 == k):
                ct +=1
            if(len(prefixmap) == 0):
                prefixmap[sum1] = 1
                continue
            if(sum1-k in prefixmap):
                ct += prefixmap[sum1-k]
            if(sum1 in prefixmap):
                prefixmap[sum1] += 1
            else:
                prefixmap[sum1] = 1
        return ct
            


